var/spack/repos/builtin/packages/mpich/package.py

```diff
@@ -89,6 +97,8 @@ class Mpich(AutotoolsPackage):
     depends_on('findutils', type='build')
     depends_on('pkgconfig', type='build')
 
+    depends_on('hwloc @1.0.0:', when='+hwloc')
```

Comment:
```suggestion
    depends_on('hwloc@2.0.0:', when='@3.3: +hwloc')
```
Since we cannot apply the patch to anything before 3.3, it also does not make sense to add a dependency then. Also, mpich does not compile with hwloc 1.x.